watertap/flowsheets/full_water_resource_recovery_facility/test/test_BSM2_P_extension.py

```diff
@@ -54,61 +55,61 @@ def test_solve(self, system_frame):
         m = system_frame
 
         assert value(m.fs.Treated.properties[0].flow_vol) == pytest.approx(
-            0.2422, rel=1e-3
+            0.24219, rel=1e-3
         )
         assert value(m.fs.Treated.properties[0].conc_mass_comp["S_A"]) == pytest.approx(
-            4.453e-06, rel=1e-3
+            6.4300e-07, rel=1e-3
         )
         assert value(m.fs.Treated.properties[0].conc_mass_comp["S_F"]) == pytest.approx(
-            0.00022875, rel=1e-3
+            0.00027610, rel=1e-3
         )
         assert value(m.fs.Treated.properties[0].conc_mass_comp["S_I"]) == pytest.approx(
             0.057450, rel=1e-3
         )
         assert value(
             m.fs.Treated.properties[0].conc_mass_comp["S_N2"]
-        ) == pytest.approx(0.024905, rel=1e-3)
+        ) == pytest.approx(0.052060, rel=1e-3)
         assert value(
             m.fs.Treated.properties[0].conc_mass_comp["S_NH4"]
-        ) == pytest.approx(0.031034, rel=1e-3)
+        ) == pytest.approx(0.00017686, rel=1e-3)
         assert value(
             m.fs.Treated.properties[0].conc_mass_comp["S_NO3"]
-        ) == pytest.approx(0, abs=1e-6)
+        ) == pytest.approx(0.0060483, rel=1e-3)
         assert value(
             m.fs.Treated.properties[0].conc_mass_comp["S_O2"]
-        ) == pytest.approx(0.0077457, rel=1e-3)
+        ) == pytest.approx(0.0076767, rel=1e-3)
         assert value(
             m.fs.Treated.properties[0].conc_mass_comp["S_PO4"]
-        ) == pytest.approx(0.7651583, rel=1e-3)
+        ) == pytest.approx(0.65267, rel=1e-3)
         assert value(m.fs.Treated.properties[0].conc_mass_comp["S_K"]) == pytest.approx(
-            0.37011, rel=1e-2
+            0.36810, rel=1e-3
         )
         assert value(
             m.fs.Treated.properties[0].conc_mass_comp["S_Mg"]
-        ) == pytest.approx(0.0182625, rel=1e-3)
+        ) == pytest.approx(0.018654, rel=1e-3)
         assert value(
             m.fs.Treated.properties[0].conc_mass_comp["S_IC"]
-        ) == pytest.approx(0.149959, rel=1e-3)
+        ) == pytest.approx(0.15148, rel=1e-3)
         assert value(
             m.fs.Treated.properties[0].conc_mass_comp["X_AUT"]
-        ) == pytest.approx(0, abs=1e-6)
+        ) == pytest.approx(0.00041597, rel=1e-3)
```

Comment:
Seems questionable that a non-negligible amount of X_AUT would end up in the treated effluent. I wonder if any existing results from the literature would help us verify whether a non-negligible amount of X_AUT in the treated effluent is a reasonable expectation here.

Reply:
From what I test, the previous zero result is unstable and tests cannot pass on github even though it is an optimal solution local. And I believe the current result is correct because if you check the output of ASM2d_ADM1_translator, the zero_flow_components are all reset to 1e-10 as we coded in the translator block, the previous results are not